### PR TITLE
Merge master to v1.0-beta

### LINKED ISSUE #456: Horizontal scrolling is buggy on mobile
## Expected Behavior
On mobile devices, the user should be able to scroll horizontally within the table.

## Current Behavior
The table only scrolls forwards, never backwards, and even then, it's quite laggy.

## Steps to Reproduce (for bugs)
Use Chrome's device emulator here http://schrodinger.github.io/fixed-data-table-2/example-object-data.html

## Context
Our users aren't able to scroll the table horizontally on mobile devices.

## Your Environment
Any environment can be used, I think, since the examples are behaving exactly as in our application.

### PR DESCRIPTION
**Pull request by pradeepnschrodinger**
## Description, Motivation and Context

Merges latest master to v1.0-beta to prepare for 1.0-beta production release.

Improvements/bug fixes that gets merged from master:
* Add roles and aria-attributes for initial screen reader support (#445) (Helps #440)
* Fix table scrolling while touch scroll is disabled (#458) (Fixes #456)

## How Has This Been Tested?

Tested on our examples and verified the role attributes and touch scrolling.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: src/FixedDataTable.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ import PropTypes from 'prop-types';
 import ReactTouchHandler from 'ReactTouchHandler';
 import ReactWheelHandler from 'ReactWheelHandler';
 import Scrollbar from 'Scrollbar';
+import ariaAttributesSelector from 'ariaAttributes';
```
Comment (pradeepnschrodinger) [inline src/FixedDataTable.js]: I put this to a selector. Will be easier to manage in the near future.

File: src/FixedDataTable.js [hunk 2]
```diff
@@ -593,11 +594,13 @@ class FixedDataTable extends React.Component {
       this._wheelHandler.onWheel,
       { passive: false }
     );
-    this._divRef && this._divRef.addEventListener(
-      'touchmove',
-      this._touchHandler.onTouchMove,
-      { passive: false }
-    );
+    if (this.props.touchScrollEnabled) {
```
Comment (pradeepnschrodinger) [inline src/FixedDataTable.js]: only add the touch event handler if the prop is on

File: src/FixedDataTable.js [hunk 3]
```diff
@@ -816,11 +833,13 @@ class FixedDataTable extends React.Component {
...
           cx('public/fixedDataTable/main'),
         )}
+        role="grid"
+        aria-rowcount={ariaRowCount}
         tabIndex={tabIndex}
         onKeyDown={this._onKeyDown}
-        onTouchStart={this._touchHandler.onTouchStart}
-        onTouchEnd={this._touchHandler.onTouchEnd}
-        onTouchCancel={this._touchHandler.onTouchCancel}
+        onTouchStart={touchScrollEnabled ? this._touchHandler.onTouchStart : null}
```
Comment (pradeepnschrodinger) [inline src/FixedDataTable.js]: don't register touch event handlers unless prop is on

File: src/Scrollbar.js [hunk 4]
```diff
@@ -512,11 +512,11 @@ class Scrollbar extends React.PureComponent {
   }
 
   getTouchX = (/*object*/ e) => {
-    return Math.round(e.targetTouches[0].pageX - e.target.getBoundingClientRect().x);
+    return Math.round(e.targetTouches[0].clientX - e.target.getBoundingClientRect().x);
```
Comment (pradeepnschrodinger) [inline src/Scrollbar.js]: fixes touch positioning of the scrollbar thumb

File: src/selectors/ariaAttributes.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ * @param {boolean} useFooter
+ * @return {{
+ *   ariaGroupHeaderIndex: number,
+ *   ariaHeaderIndex: number,
+ *   ariaFooterIndex: number,
+ *   ariaRowCount: number,
+ *   ariaRowIndexOffset: number
+ * }}
+ */
+function calculateAriaAttributes(rowsCount, useGroupHeader, useFooter) {
```
Comment (pradeepnschrodinger) [inline src/selectors/ariaAttributes.js]: @Tanner-MS , can you review this function? I just copied your PR for master, and renamed/rearranged a bit.

Comment (Tanner-MS) [reply src/selectors/ariaAttributes.js]: Looks good

**Pull request merged**